### PR TITLE
Feature/159 refactor navbar

### LINKED ISSUE #159: Refactor navbar to Vuetify
**DoD**
The navbar should be written with Vuetify. 

## Context
Relates to #144 
Talk to @itggot-TE4/blue-team regarding help with migrating other components to Vuetify that is dependant on navbar.

### PR DESCRIPTION
**Pull request by te4-gabriel-stark**
## What issue did you implement or fix?
Fixed #159 

## Summary
- Converted from bootstrap to vuetify in HeaderNav.
- Changed the name of HeaderNav to NavbarComponent to follow code style.
- Updated the GoogleLogin file aswell.

## Type of change
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## How can this implementation be improved?
- Make the test pass.

### CODE REVIEW

File: frontend/package.json [hunk 1]
```diff
@@ -22,6 +22,7 @@
     "jsbarcode": "^3.11.0",
     "jszip": "^3.1.5",
     "lodash": "^4.17.15",
+    "tslint": "^5.20.1",
```
Comment (te4-david-petersson) [inline frontend/package.json]: this should not be included

File: frontend/public/index.html [hunk 2]
```diff
@@ -10,6 +10,7 @@
     <title>yabs</title>
     <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:100,300,400,500,700,900">
     <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@mdi/font@latest/css/materialdesignicons.min.css">
+    <link href="https://unpkg.com/tailwindcss@^1.0/dist/tailwind.min.css" rel="stylesheet">
```
Comment (te4-david-petersson) [inline frontend/public/index.html]: what does this library do?

File: frontend/src/api/loans.ts [hunk 3]
```diff
@@ -18,10 +18,9 @@ export default {
 
   },
   async create(request: any) {
-
     try {
       const response = await axios.post(`http://localhost:3000/api/v1/loans`, request,
-        {headers, withCredentials: true});
+        {headers});
```
Comment (te4-david-petersson) [inline frontend/src/api/loans.ts]: why is this removed? Aren't cookies used for authentication?

File: frontend/src/api/users.ts [hunk 4]
```diff
@@ -16,6 +16,20 @@ export default {
       return Promise.reject(error);
     }
 
+  },
+  async get(id: string) {
```
Comment (te4-david-petersson) [inline frontend/src/api/users.ts]: We should open up a issue for using ES6 classes instead similar to rate-at-arms project from Linus.

File: frontend/src/components/GoogleLogin.vue [hunk 5]
```diff
@@ -1,7 +1,10 @@
-<template lang="pug">
-  div
-    a.text-warning(v-if="usersModule.currentUser" href="#" @click="signOut();") SIGN OUT
-    div#signin2
+<template>
+    <v-btn class="ml-2" text color="grey">
+      <div id="signin2"></div>
+    </v-btn>
+
+    <!-- div#signin2(v-else="usersModule.currentUser") -->
```
Comment (te4-david-petersson) [inline frontend/src/components/GoogleLogin.vue]: Can this be removed?

Comment (te4-david-petersson) [reply frontend/src/components/GoogleLogin.vue]: 👎

File: frontend/src/components/GoogleLogin.vue [hunk 6]
```diff
@@ -13,17 +16,25 @@ export default class GoogleLogin extends Vue {
...
+      width: 200,
       height: 50,
-      longtitle: true,
+      longtitle: false,
       theme: 'dark',
       onsuccess: this.onSignIn,
       onfailure: this.onFailure,
     });
   }
+  public updated() {
```
Comment (te4-david-petersson) [inline frontend/src/components/GoogleLogin.vue]: when is updated called?

File: frontend/src/components/GoogleLogin.vue [hunk 7]
```diff
@@ -43,16 +54,16 @@ export default class GoogleLogin extends Vue {
 
   public onFailure(error: any) {
     // TODO: show this in a notification
-    console.error(error);
+    console.error(error, 'hej');
```
Comment (te4-david-petersson) [inline frontend/src/components/GoogleLogin.vue]: is this necessary?

File: frontend/src/components/LoaningForm.vue [hunk 8]
```diff
@@ -18,7 +18,7 @@ import UsersModule from '../store/modules/UsersModule';
 
 interface Form {
   lent_by_id: any;
-  loaned_by_id: '';
+  loaned_by_id: any;
```
Comment (te4-david-petersson) [inline frontend/src/components/LoaningForm.vue]: shouldn't this be a number?

File: frontend/src/components/LoaningForm.vue [hunk 9]
```diff
@@ -34,6 +34,8 @@ export default class LoaningForm extends Vue {
   public onSubmit(evt: Event) {
     evt.preventDefault();
     this.form.lent_by_id = UsersModule.currentUser ? UsersModule.currentUser.uid : '';
+    this.form.loaned_by_id = parseInt(this.form.loaned_by_id, 10);
+    console.log(this.form.loaned_by_id);
```
Comment (te4-david-petersson) [inline frontend/src/components/LoaningForm.vue]: redundant console.log

File: frontend/src/components/LoaningForm.vue [hunk 10]
```diff
@@ -34,6 +34,8 @@ export default class LoaningForm extends Vue {
   public onSubmit(evt: Event) {
     evt.preventDefault();
     this.form.lent_by_id = UsersModule.currentUser ? UsersModule.currentUser.uid : '';
+    this.form.loaned_by_id = parseInt(this.form.loaned_by_id, 10);
```
Comment (te4-david-petersson) [inline frontend/src/components/LoaningForm.vue]: why is this a string from the beginning?

File: frontend/src/components/NavbarComponent.vue [hunk 11]
```diff
@@ -0,0 +1,100 @@
...
+import UsersModule from '../store/modules/UsersModule';
+
+@Component({
+  components: {
+    GoogleLogin,
+  },
+})
+export default class NavbarComponent extends Vue {
+  public usersModule = UsersModule;
+  // private links = [
```
Comment (te4-david-petersson) [inline frontend/src/components/NavbarComponent.vue]: can these comments be removed?

File: frontend/src/components/RecentLoan.vue [hunk 12]
```diff
@@ -29,9 +29,7 @@ export default class RecentLoan extends Vue {
   }
 
   public created() {
-    LoansModule.fetchAll()
-      .then(() => this.$emit('loans-loaded', LoansModule.all))
-      .catch((failure: any) => console.log(failure));
+    LoansModule.fetchAll();
```
Comment (te4-david-petersson) [inline frontend/src/components/RecentLoan.vue]: Is this no longer necessary?

File: frontend/package.json (hunk 1 above)
Comment (te4-gabriel-stark) [reply frontend/package.json]: 👍

File: frontend/public/index.html (hunk 2 above)
Comment (te4-gabriel-stark) [reply frontend/public/index.html]: This is some library ythat arvid added, dont know what it does.

File: frontend/src/api/loans.ts (hunk 3 above)
Comment (te4-gabriel-stark) [reply frontend/src/api/loans.ts]: True

File: frontend/src/components/GoogleLogin.vue (hunk 5 above)
Comment (te4-gabriel-stark) [reply frontend/src/components/GoogleLogin.vue]: If you mean the comment, then yes.

File: frontend/src/components/GoogleLogin.vue (hunk 6 above)
Comment (te4-gabriel-stark) [reply frontend/src/components/GoogleLogin.vue]: This is a automatic function that vue has, it runs when something changes.

File: frontend/src/components/GoogleLogin.vue (hunk 7 above)
Comment (te4-gabriel-stark) [reply frontend/src/components/GoogleLogin.vue]: nope

File: frontend/src/components/LoaningForm.vue (hunk 8 above)
Comment (te4-gabriel-stark) [reply frontend/src/components/LoaningForm.vue]: It get the input as a string so you could type number | string instead of any.

File: frontend/src/components/LoaningForm.vue (hunk 10 above)
Comment (te4-gabriel-stark) [reply frontend/src/components/LoaningForm.vue]: See comment above

File: frontend/src/components/LoaningForm.vue (hunk 9 above)
Comment (te4-gabriel-stark) [reply frontend/src/components/LoaningForm.vue]: yup

File: frontend/src/components/NavbarComponent.vue (hunk 11 above)
Comment (te4-gabriel-stark) [reply frontend/src/components/NavbarComponent.vue]: yes

File: frontend/src/components/RecentLoan.vue (hunk 12 above)
Comment (te4-gabriel-stark) [reply frontend/src/components/RecentLoan.vue]: The function has been rewritten so it does not need the other stuff.

File: frontend/src/components/GoogleLogin.vue [hunk 13]
```diff
@@ -1,13 +1,8 @@
...
-      class="text-warning"
-      href="#"
-      @click="signOut"
-    >SIGN OUT</a>
-    <div id="signin2" />
-  </div>
+  <v-btn class="ml-2" text color="grey">
+    <div id="signin2"></div>
+  </v-btn>
+  <!-- div#signin2(v-else="usersModule.currentUser") -->
```
Comment (TE4-oskar-pilborg) [inline frontend/src/components/GoogleLogin.vue]: I'm just curious, why did we take the v-if away when it is used to show log out link when logged in?

Comment (te4-gabriel-stark) [reply frontend/src/components/GoogleLogin.vue]: look in the navbarcomponent

File: frontend/src/views/Profile.vue [hunk 14]
```diff
@@ -1,30 +1,16 @@
 <template>
-  <div>
-    <p v-if="!usersModule.currentUser">
-      Du är inte inloggad
-    </p>
-    <div
-      v-else
-      class="view"
+  <v-container>
+    <h1>{{usersModule.currentUser.name}} - {{usersModule.currentUser.klass}}</h1>
```
Comment (TE4-oskar-pilborg) [inline frontend/src/views/Profile.vue]: Why did you change a file that navbar isn't dependent on?

File: frontend/src/components/GoogleLogin.vue (hunk 13 above)
Comment (TE4-Mattis-Abrahamsson) [inline frontend/src/components/GoogleLogin.vue]: Is this needed?

File: frontend/src/components/NavbarComponent.vue [hunk 15]
```diff
@@ -0,0 +1,100 @@
...
+        </v-btn>
+      </v-toolbar-items>
+      <v-spacer />
+      <v-toolbar-items v-if="usersModule.userState && usersModule.userState.currentUser">
+        <v-btn :to="'/users/'+ usersModule.currentUser.uid" text>
+          <v-icon class="px-3">mdi-settings</v-icon>Profil
+        </v-btn>
+        <v-btn to="/add" text>
+          <v-icon class="px-3">add</v-icon>Add
+        </v-btn>
```
Comment (TE4-Mattis-Abrahamsson) [inline frontend/src/components/NavbarComponent.vue]: change to /admin

File: frontend/src/api/users.ts (hunk 4 above)
Comment (te4-gabriel-stark) [reply frontend/src/api/users.ts]: :)

File: frontend/src/views/Profile.vue (hunk 14 above)
Comment (te4-gabriel-stark) [reply frontend/src/views/Profile.vue]: It's the profile view so now it looks better :), but if you want more clarity ask @TE4-arvid-nilber

File: frontend/src/components/GoogleLogin.vue (hunk 13 above)
Comment (te4-gabriel-stark) [reply frontend/src/components/GoogleLogin.vue]: Fixed!

File: frontend/src/components/NavbarComponent.vue (hunk 15 above)
Comment (te4-gabriel-stark) [reply frontend/src/components/NavbarComponent.vue]: yes

**Comment by te4-gabriel-stark** [comment]
It seems that the userModule have stopped working in this branch, but i have heard that @TE4-oskar-pilborg fixed it in #201

File: frontend/src/components/LoaningForm.vue [hunk 16]
```diff
@@ -51,17 +51,17 @@ import LoansModule from '../store/modules/LoansModule';
 import UsersModule from '../store/modules/UsersModule';
 
 interface Form {
-  lent_by_id: any;
-  loaned_by_id: '';
-  book_id: '';
+  lent_by_id: number | string;
+  loaned_by_id: number | string;
+  book_id: number | string;
```
Comment (TE4-oskar-pilborg) [inline frontend/src/components/LoaningForm.vue]: why can it be both?

Comment (te4-gabriel-stark) [reply frontend/src/components/LoaningForm.vue]: It takes the input as a string, but you want to be able to change it to a number. I understand that it's confusing.

File: frontend/src/store/modules/LoansModule.ts [hunk 17]
```diff
@@ -102,6 +98,7 @@ class LoansModule extends VuexModule {
     Vue.delete(this.loanState.loans, loanId);
   }
   @Mutation
+
```
Comment (te4-david-petersson) [inline frontend/src/store/modules/LoansModule.ts]: Doesn't Mutation have to be directly on top of the function?

File: frontend/src/views/Add.vue [hunk 18]
```diff
@@ -62,6 +62,7 @@ import RecentBook from '@/components/RecentBook.vue';
 import RecentLoan from '@/components/RecentLoan.vue';
 import RecentTitel from '@/components/RecentTitel.vue';
 import CanvasContainer from '@/components/CanvasContainer.vue';
+import TitelForm from '@/components/TitelForm.vue';
 import TitelFormComponent from '@/components/TitelFormComponent.vue';
```
Comment (te4-david-petersson) [inline frontend/src/views/Add.vue]: Isn't TitelForm.vue removed and replaced with corresponding component? Also swenglish :) TitleForm, might be nicer

Comment (TE4-Mattis-Abrahamsson) [reply frontend/src/views/Add.vue]: Add.vue is removed in base branch
